lib/stages/HFBAccumulate.cpp

```diff
@@ -97,64 +97,60 @@ HFBAccumulate::HFBAccumulate(Config& config_, const std::string& unique_name,
 
 HFBAccumulate::~HFBAccumulate() {}
 
-void HFBAccumulate::init_first_frame(float* input_data, float* sum_data,
-                                     const uint32_t in_buffer_ID) {
+void HFBAccumulate::init_first_frame(HFBFrameView& in_frame, float* sum_data) {
 
     int64_t fpga_seq_num_start =
         fpga_seq_num_end - (_num_frames_to_integrate - 1) * _samples_per_data_set;
-    memcpy(sum_data, input_data, _num_frb_total_beams * _factor_upchan * sizeof(float));
-    total_lost_timesamples += get_fpga_seq_start_hfb(in_buf, in_buffer_ID) - fpga_seq_num_start;
+    memcpy(sum_data, in_frame.hfb.data(), _num_frb_total_beams * _factor_upchan * sizeof(float));
+    total_lost_timesamples += in_frame.fpga_seq_start - fpga_seq_num_start;
     // Get the first FPGA sequence no. to check for missing frames
-    fpga_seq_num = get_fpga_seq_start_hfb(in_buf, in_buffer_ID);
+    fpga_seq_num = in_frame.fpga_seq_start;
     frame++;
 
     DEBUG("\nInit frame. fpga_seq_start: {:d}. sum_data[0]: {:f}",
-          get_fpga_seq_start_hfb(in_buf, in_buffer_ID), sum_data[0]);
+          in_frame.fpga_seq_start, sum_data[0]);
 }
 
-void HFBAccumulate::integrate_frame(float* input_data, float* sum_data,
-                                    const uint32_t in_buffer_ID) {
+void HFBAccumulate::integrate_frame(HFBFrameView& in_frame, float* sum_data) {
     frame++;
     fpga_seq_num += _samples_per_data_set;
-    total_lost_timesamples += get_fpga_seq_start_hfb(in_buf, in_buffer_ID) - fpga_seq_num;
-    fpga_seq_num = get_fpga_seq_start_hfb(in_buf, in_buffer_ID);
+    total_lost_timesamples += in_frame.fpga_seq_start - fpga_seq_num;
+    fpga_seq_num = in_frame.fpga_seq_start;
 
     // Integrates data from the input buffer to the output buffer.
-    for (uint beam = 0; beam < _num_frb_total_beams; beam++) {
-        for (uint freq = 0; freq < _factor_upchan; freq++) {
-            sum_data[beam * _factor_upchan + freq] += input_data[beam * _factor_upchan + freq];
-        }
+    for (uint32_t i = 0; i < _num_frb_total_beams * _factor_upchan; i++) {
+        sum_data[i] += in_frame.hfb.data()[i];
```

Comment:
@jrs65 I had to use `in_frame.hfb.data()[i]` instead of just `in_frame.hfb[i]` as it seg faults otherwise. Any idea what could cause this?